crates/rsonpath-lib/tests/query_parser_tests.rs

```diff
@@ -253,25 +258,39 @@ mod proptests {
 
     // .label or ['label']
     fn any_child() -> impl Strategy<Value = Selector> {
-        prop_oneof![any_member().prop_map(|x| (format!(".{x}"), x)), any_index(),].prop_map(|(s, l)| Selector {
+        prop_oneof![any_member().prop_map(|x| (format!(".{x}"), x)), any_name(),].prop_map(|(s, l)| Selector {
             string: s,
             tag: SelectorTag::Child(l),
         })
     }
 
     // ..label or ..['label']
     fn any_descendant() -> impl Strategy<Value = Selector> {
-        prop_oneof![any_member().prop_map(|x| (x.clone(), x)), any_index(),].prop_map(|(x, l)| Selector {
+        prop_oneof![any_member().prop_map(|x| (x.clone(), x)), any_name(),].prop_map(|(x, l)| Selector {
             string: format!("..{x}"),
             tag: SelectorTag::Descendant(l),
         })
     }
 
+    fn any_array_index_child() -> impl Strategy<Value = Selector> {
+        any_non_negative_array_index().prop_map(|i| Selector {
+            string: format!("[{}]", i.get_index()),
```

Comment:
minor, but is there a reason to not use the `Display` implementation here?

Reply:
The display of `NonNegativeArrayIndex` I presume?

It's to not depend on the tested module when setting up the test case. We can be 100% certain how `Display` of `u64` works, whereas the `Display` for `NonNegativeArrayIndex` is our code and it could format something differently.